include/tvm/runtime/container.h

```diff
@@ -274,6 +275,99 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
+  char* data;
+
+  /*! \brief String object which is moved from std::string container. */
+  class FromStd;
+};
+
+/*! \brief An object representing string moved from std::string. */
+class StringObj::FromStd : public StringObj {
+ public:
+  /*! \brief Container that holds the memory. */
+  std::string data_container;
+};
+
+/*! \brief reference to string objects. */
+class String : public ObjectRef {
+ public:
+  /*!
+   * \brief Construct a new String object
+   *
+   * \param other The moved std::string object
+   */
+  explicit String(std::string&& other) {
+    auto ptr = make_object<StringObj::FromStd>();
```

Comment:
We only need to define a constructor, and it should cover the move and copy constructors(when const parameter is passed, it will trigger a copy, otherwise a move.
```c++
explicit String(std::string other) {
}
```